### PR TITLE
itk: 5.0.0 -> 5.0.1

### PR DESCRIPTION
**Pull request by mmahut**
###### Motivation for this change

Updating to 5.0.1, hopefully fixes a number of builds.

Also moving the source to github.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @viric

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk ants elastix EZminc python27Packages.pyezminc

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

**Comment by mmahut** [comment]
@GrahamcOfBorg build ants

**Comment by mmahut** [comment]
@GrahamcOfBorg build elastix

**Comment by mmahut** [comment]
@GrahamcOfBorg build ezminc

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

https://hydra.nixos.org/build/98720007

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

**Comment by mmahut** [comment]
The build is too long for ofborg, but I success for me locally. Anyone willing to run nix-review on it, please.

File: pkgs/development/libraries/itk/default.nix [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 stdenv.mkDerivation rec {
-  name = "itk-5.0.0";
+  pname = "itk";
+  version = "5.0.1";
 
-  src = fetchurl {
-    url = mirror://sourceforge/itk/InsightToolkit-5.0.0.tar.xz;
-    sha256 = "0bs63mk4q8jmx38f031jy5w5n9yy5ng9x8ijwinvjyvas8cichqi";
+  src = fetchFromGitHub {
```
Comment (aanderse) [inline pkgs/development/libraries/itk/default.nix]: Please include proof this is a legitimate change so we have a paper trail.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/itk/default.nix (hunk 1 above)
Comment (mmahut) [reply pkgs/development/libraries/itk/default.nix]: Their github repository is listed at the bottom on this page https://itk.org/ITK/resources/software.html

Comment (mmahut) [reply pkgs/development/libraries/itk/default.nix]: Also, the current release seems to be only on github.

**Comment by aanderse** [comment]
`~/nixpkgs> nix-review pr 66567`
...
```
https://github.com/NixOS/nixpkgs/pull/66567
2 package are marked as broken and were skipped:
c3d seg3d

5 package failed to build:
ants elastix ezminc python27Packages.pyezminc simpleitk

1 package were build:
itk
```
:man_shrugging:

**Comment by mmahut** [comment]
Odd.

@GrahamcOfBorg build simpleitk

**Comment by mmahut** [comment]
@GrahamcOfBorg build simpleitk

**Comment by risicle** [comment]
Fails on macos 10.13:

```
builder for '/nix/store/l9g39yk5rmqmpvwrp3pcc0342s8av45g-simpleitk-1.2.2.drv' failed with exit code 2; last 10 log lines:
  [ 97%] Swig source to generate TCL wrapping
  Scanning dependencies of target SimpleITKTclsh
  [ 97%] Building CXX object Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/SimpleITKTCL_wrap.cxx.o
  /tmp/nix-build-simpleitk-1.2.2.drv-0/source/build/Wrapping/Tcl/SimpleITKTCL_wrap.cxx:166:10: fatal error: 'tcl.h' file not found
  #include <tcl.h>
           ^~~~~~~
  1 error generated.
  make[2]: *** [Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/build.make:407: Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/SimpleITKTCL_wrap.cxx.o] Error 1
  make[1]: *** [CMakeFiles/Makefile2:2554: Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/all] Error 2
  make: *** [Makefile:152: all] Error 2
```

**Comment by mmahut** [comment]
> Fails on macos 10.13:
That's odd. It builds fine under GrahamcOfBorg. Do you mind trying again please?

**Comment by mmahut** [comment]
ping @bcdarwin do you mind taking a look at ```ants elastix ezminc python27Packages.pyezminc simpleitk```?

**Comment by mmahut** [comment]
@risicle do you mind testing again on darwin?

**Comment by mmahut** [comment]
@aanderse can you please try again? I get

```
2 package are marked as broken and were skipped:
c3d seg3d

4 package failed to build:
ants elastix ezminc python27Packages.pyezminc

2 package were build:
itk simpleitk
```

**Comment by risicle** [comment]
> @risicle do you mind testing again on darwin?

I currently am, but it takes a while...

**Comment by aanderse** [comment]
@mmahut yes, identical results now:
```
https://github.com/NixOS/nixpkgs/pull/66567
2 package are marked as broken and were skipped:
c3d seg3d

4 package failed to build:
ants elastix ezminc python27Packages.pyezminc

2 package were build:
itk simpleitk
```

**Comment by mmahut** [comment]
@aanderse do you mind sharing your log? I did the nix-review of this on two separate machines and simpleitk did build for me.

**Comment by risicle** [comment]
Getting closer on macos now we're including `tcl` and `tk`:

```
  [ 97%] Built target SimpleITKLuaModule_LUA
  [ 97%] Swig source to generate TCL wrapping
  Scanning dependencies of target SimpleITKTclsh
  [ 97%] Building CXX object Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/SimpleITKTCL_wrap.cxx.o
  [ 97%] Linking CXX executable bin/SimpleITKTclsh
  ld: framework not found tcl
  clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/build.make:555: Wrapping/Tcl/bin/SimpleITKTclsh] Error 1
  make[1]: *** [CMakeFiles/Makefile2:2554: Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/all] Error 2
  make: *** [Makefile:152: all] Error 2
```

I'm trying a rebuild right now with `darwin.apple_sdk.frameworks.Tcl` included...

**Comment by risicle** [comment]
Getting closer...

```
Scanning dependencies of target SimpleITKTclsh
[ 97%] Building CXX object Wrapping/Tcl/CMakeFiles/SimpleITKTclsh.dir/SimpleITKTCL_wrap.cxx.o
[ 97%] Linking CXX executable bin/SimpleITKTclsh
Undefined symbols for architecture x86_64:
  "_Tcl_MainEx", referenced from:
      _main in SimpleITKTCL_wrap.cxx.o
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by risicle** [comment]
Hmm, yes, same error as before. If it builds on hydra, I guess that might be enough - but I suspect there's some impurity at play here.

**Comment by mmahut** [comment]
@risicle how do you build it? using ``nix-review``?

**Comment by risicle** [comment]
both `nix-review` and `nix-build` - same results.

**Comment by mmahut** [comment]
Odd, I'm going to merge it, as it passes fine on darwin. I have mailed the maintainer for other packages as well, we will mark them as broken if no reply.

**Pull request merged**